.github/workflows/actions/build-core/action.yml

```diff
@@ -0,0 +1,28 @@
+name: 'Build Stencil Core'
+description: 'builds the core compiler'
+runs:
+  using: composite
+  steps:
+    # since this job runs on its own virtual machine, we need to get the dependencies (which may be cached for us)
+    - name: Get Core Dependencies
+      uses: ./.github/workflows/actions/get-core-dependencies
+
+    - name: Build
+      run: npm run build -- --ci
+      shell: bash
+
+    - name: Validate Build
+      run: npm run test.dist
+      shell: bash
+
+    - name: Validate Testing
+      run: npm run test.testing
+      shell: bash
+
+    - name: Upload Build Artifacts
+      uses: ./.github/workflows/actions/upload-archive
+      with:
+        name: stencil-core
+        output: stencil-core-build.zip
+        paths: build cli compiler dependencies.json dev-server internal mock-doc scripts screenshot sys testing
```

Comment:
I'm not sure if it's feasible, but if the artifact is generated using `npm pack` instead we don't have to keep this list in sync with `package.json:files` and it would ensure that the tests run against the to-be-published package (or at least close it).

Reply:
That's fair, I'm gonna create a "throwaway PR" to test this (so that I don't spam folks by verifying this and other changes) later this week and will report back

Reply:
I tried this with a `mv \`npm pack\` some-tar-name.tar` - I ran into problems a few problems here:
- `npm pack` packed everything, including `node_modules/`. This wasn't a huge time sink (although there was some), but it definitely spammed the logs when I unpacked it in a subsequent step
- When unpacking `npm pack`, GHA would crash. No message/logs (to be fair, I didn't enable extra debugging in the repo), just an exit code of 1.  

I may poke at this a little more so we don't have to update these paths, but the early experiments weren't super encouraging 😆

Reply:
Unfortunately I think we have to remain on this implementation - I couldn't get everything packed up the way I wanted to that didn't involve one list of directories or another to maintain :-(